README.md

```diff
@@ -37,15 +37,63 @@ Available frontends to define the folded circuit:
 - [arkworks](https://github.com/arkworks-rs), arkworks contributors
 - [Circom](https://github.com/iden3/circom), iden3, 0Kims Association
 
-
 ## Usage
 
+Detailed usage and design documentation can be found [here](https://privacy-scaling-explorations.github.io/sonobe-docs/).
+
+With sonobe, developers can choose which folding scheme and decider to use. The following code snippet shows how to fold a circuit using the Nova folding scheme and an EVM-compatible decider:
+
+```rust
+// Initialize Nova as our IVC scheme
+// `BTCBlockCheckerFCircuit` is an R1CS checking the validity of a bitcoin block
+// This example is a provable light client for Bitcoin
+type NOVA = Nova<
+    Projective, // projective first curve points (required to form a curve cycle with `Projective2`)
+    GVar, // R1CS representation of the first curve points
+    Projective2, // projective second curve points
+    GVar2, // R1CS representation of second curve points
+    BTCBlockCheckerFCircuit<Fr>, // circuit being folded
+    KZG<'static, Bn254>, // sonobe is agnostic to the homomorphic commitment scheme used 
+    Pedersen<Projective2>, // sonobe is agnostic to the homomorphic commitment scheme used 
+>;
+
+// An EVM-compatible decider for our IVC proof, combining a KZG and a groth16 proof together 
+type DECIDER = Decider<
+    Projective,
+    GVar,
+    Projective2,
+    GVar2,
+    BTCBlockCheckerFCircuit<Fr>,
+    KZG<'static, Bn254>,
+    Pedersen<Projective2>,
+    Groth16<Bn254>, // to be evm-friendly, the final decider combines a groth16 proof with multiple kzg openings
+    NOVA,
+>;
+
+// Initializing Nova with the circuit and the initial state
+let mut nova = NOVA::init(&prover_params, circuit, z_0.clone()).unwrap();
+
+// Computing incremental steps, i.e. recursively verifying bitcoin blocks
+for i in 0..n_blocks_checked {
+    nova.prove_step().unwrap();
+}
+```
+
+See the full example for using sonobe to recursively check bitcoin blocks [here](https://github.com/dmpierre/folding-schemes-light-btc/tree/main).
+
+### Folding Schemes and IVC
+
+A folding scheme reduces the task of checking multiple instances in some relation to the task of checking a single one. The canonical definition of a folding scheme entailed computing a folded instance as a random linear combination of the original instances. 
+
+A folding scheme can also be used as a primitive to achieve incrementally verifiable computation (IVC). One such example is [Nova](https://eprint.iacr.org/2021/370.pdf), which showed an IVC scheme with interesting performance properties. Note that an IVC proof is neither succint nor zero-knowledge. Hence, a final ("decider") zkSNARK will be required to prove the correctness of an IVC result.
```

Comment:
maybe this is a 'duplicate' of the other "Folding Schemes introduction" section below? not that needs to be removed but maybe we could merge both of them into a single one?